go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1111,12 +1117,26 @@ func (bnc *BaseNetworkController) addNetworkPolicy(policy *knet.NetworkPolicy) e
 
 	// 1. save required namespace information to use for netpol create,
 	npKey := getPolicyKey(policy)
-	nsInfo, nsUnlock := bnc.getNamespaceLocked(policy.Namespace, true)
-	if nsInfo == nil {
-		return fmt.Errorf("unable to get namespace for network policy %s: namespace doesn't exist", npKey)
+
+	var aclLogging ACLLoggingLevels
+	if bnc.doesNetworkRequireIPAM() {
+		nsInfo, nsUnlock := bnc.getNamespaceLocked(policy.Namespace, true)
+		if nsInfo == nil {
+			return fmt.Errorf("unable to get namespace for network policy %s: namespace doesn't exist", npKey)
+		}
+		aclLogging = nsInfo.aclLogging
+		nsUnlock()
+	} else {
+		var (
+			err      error
+			nsUnlock func()
+		)
+		aclLogging, nsUnlock, err = bnc.loadNamespaceAclLoggingConfig(policy.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get nsInfo for namespace %s", policy.Namespace)
+		}
+		nsUnlock()
```

Comment:
maybe it makes sense to create a function like "getNamespaceLogLevel" to encapsulate this logic?